From c4d228323c9c302b355cfd27cd59809110d7c74e Mon Sep 17 00:00:00 2001
From: Grigori Goronzy <greg@kinoho.net>
Date: Wed, 9 Feb 2022 22:23:11 +0100
Subject: [PATCH] dtoverlays: add rgb565-padhi output to
 vc4-kms-dpi-generic

This mode is used by some DPI based display adapters, such as the
'VGA565' and 'VGA Zero' low-cost VGA adapters for the Raspberry Pi.

Signed-off-by: Grigori Goronzy <greg@kinoho.net>
---
 arch/arm/boot/dts/overlays/README                          | 2 ++
 arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts | 2 ++
 2 files changed, 4 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3649,6 +3649,8 @@ Params: clock-frequency         Display
         width-mm                Define the screen width in mm
         height-mm               Define the screen height in mm
         rgb565                  Change to RGB565 output on GPIOs 0-19
+        rgb565-padhi            Change to RGB565 output on GPIOs 0-8, 12-17, and
+                                20-24
         rgb666-padhi            Change to RGB666 output on GPIOs 0-9, 12-17, and
                                 20-25
         rgb888                  Change to RGB888 output on GPIOs 0-27
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
@@ -65,6 +65,8 @@
 
 		rgb565 = <&panel_generic>, "bus-format:0=0x1017",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_16bit_gpio0>;
+		rgb565-padhi = <&panel_generic>, "bus-format:0=0x1020",
+			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_16bit_cpadhi_gpio0>;
 		rgb666-padhi = <&panel_generic>, "bus-format:0=0x1015",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_18bit_cpadhi_gpio0>;
 		rgb888 = <&panel_generic>, "bus-format:0=0x100a",
